coursier.bzl

```diff
@@ -575,6 +667,22 @@ def _coursier_fetch_impl(repository_ctx):
         False,  # not executable
     )
 
+    # If this is an unpinned repository, expose the script to let users pin the
+    # state of the fetch in `<workspace_root>/maven_install.json`.
+    #
+    # $ bazel run @unpinned_maven//:pin
+    #
+    if not repository_ctx.attr.maven_install_json:
```

Comment:
Yes. `maven_install.json` must only be generated from `@unpinned_maven`, since `@unpinned_maven` doesn't depend on `maven_install` to prevent cyclic dependencies. 

`@unpinned_maven//:pin` -> generates `maven_install.json` -> generates `@maven//...` targets.